File: create_large_file.cpp
#include <fstream>
#include <iostream>
#include <random>
#include <string>

std::string generate_random_string(size_t length) {
  const std::string characters =
      "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
  std::string random_string;
  std::random_device rd;
  std::mt19937 generator(rd());
  std::uniform_int_distribution<> distribution(0, characters.size() - 1);

  for (size_t i = 0; i < length; ++i) {
    random_string += characters[distribution(generator)];
  }

  return random_string;
}

int main() {
  const long long fileSize = 1'000'000'000;  // 1 GB file
  std::string filePath = "./large_file_1_gb.txt";

  std::ofstream filePtr(filePath, std::ios::out);
  if (!filePtr.is_open()) {
    std::cerr << "Failed to open file for writing." << std::endl;
    return 1;
  }

  const int chunkSize =
      10'000'000;  // doing chunking to reduce the number of disk access
  for (int i = 0; i < fileSize / chunkSize; ++i) {
    std::string chunk = generate_random_string(chunkSize);
    filePtr.write(chunk.c_str(), chunk.size());
  }

  filePtr.close();
  return 0;
}

File: create_large_file.py
import string
import random


def generate_random_string(length):
    characters = string.ascii_letters + string.digits
    random_string = ''.join(random.choice(characters) for _ in range(length))
    return random_string


def main():
	file_size = 1_000_000_000  # 1 GB file
	file_path = f'./large_file_{file_size//1_000_000_000}_gb.txt'

	with open(file_path, 'w+') as file_ptr:
		chunk_size = 10_000_000  # doing chunking to reduce the number of disk access
		for _ in range(file_size//chunk_size):
			chunk = generate_random_string(chunk_size)
			file_ptr.write(chunk)


if __name__ == '__main__':
	main()
